### PR TITLE
Experiment/finetuning

### PR DESCRIPTION
**Pull request by JeroenMandersloot**
* Renamed `BaseModel` to `Architecture` to make its intended usage more explicit.
* `Architecture` can now also be used to easily load a `ScorerModel`, i.e. a model that implements the sklearn interface to output dissimilarity scores between pairs of images. The philosophy behind this `Architecture` class is as follows:
  - `Architecture` itself should not be responsible for making predictions of any kind, it is just a convenient way to specify a backbone architecture such as VGGFace or FaceNet and then build full-fledged downstream models from them. Only those downstream models should have an interface that lets you make predictions with them (whether that be computing an embedding or scoring pairs of images).
  - Instead of having to think beforehand about what kind of model you might need later for a particular downstream task (e.g. finetuning, embedding or scoring), you can instead just pass along a `Architecture` instance everywhere in the application. This greatly reduces cognitive load, as you don't have to constantly think about how the downstream models are implemented, what arguments they take, etc.
  - `Architecture` then gives easy access to all different types of models that you can build only when you actually need them:
    - `get_embedding_model()` returns a Keras model that can create embeddings for images.
    - `get_triplet_embedding_model()` returns a `TripletEmbeddingModel` (subclass of a Keras model) that can embed triplets and compute a triplet loss.
    - `get_scorer_model()` returns a new `ScorerModel` instance that can compute scores between pairs of images using an sklearn-like API.
  - `Architecture` now also implements a `resolution` property that returns the spatial input dimensions required by the model as a `(height, width)` tuple. So for example, you can now easily request the resolution required by any VGGFace downstream model as follows: `Architecture.VGGFACE.resolution`
* Fixed `finetuning.py` to use new data structures
* Created a sanity check for finetuning in `finetuning.py`
* Created a script to perform a learning rate test on a dataset of choice in `learning_rate_test.py`
* Changed the return format of `to_array()`
* Added unit tests for some datasets and helper functions in `data.py` such as `make_pairs()`, `make_triplets()` and `to_array()`

### CODE REVIEW

File: lr_face/models.py [hunk 1]
```diff
@@ -107,70 +167,61 @@ def load_weights(self, filepath, by_name=False):
...
-        if np.array(X).shape[1] != 2:
-            raise ValueError(
-                f'Should get n pairs, but second dimension is {np.array(X).shape[1]}')
-        return np.random.random((len(X), 2))
-
-    def __str__(self):
-        return 'Dummy'
-
-
 class BaseModel(Enum):
```
[the remark this answers is not in the document]
Comment (JeroenMandersloot) [reply lr_face/models.py:170]: Done

File: finetuning.py [hunk 2]
```diff
@@ -1,69 +1,80 @@
...
     """
 
-    anchors, positives, negatives = zip(*[(
-        triplet.anchor,
-        triplet.positive,
-        triplet.negative
-    ) for triplet in triplets])
-
     model.compile(
-        optimizer=Adam(learning_rate=3e-4),  # TODO: default choice
```
[the remark this answers is not in the document]
Comment (JeroenMandersloot) [reply finetuning.py:37]: Done
